lib/iris/tests/unit/common/metadata/test_CellMeasureMetadata.py

```diff
@@ -208,277 +205,279 @@ def setUp(self):
         qualname_combine = _qualname(self.cls.combine)
-        self.assertIn(qualname_combine, _LENIENT)
-        self.assertTrue(_LENIENT[qualname_combine])
-        self.assertTrue(_LENIENT[self.cls.combine])
-
-    def test_lenient_default(self):
-        other = sentinel.other
-        return_value = sentinel.return_value
-        with mock.patch.object(
+        assert qualname_combine in _LENIENT
+        assert _LENIENT[qualname_combine]
+        assert _LENIENT[self.cls.combine]
+
+    def test_lenient_default(self, mocker):
+        other = mocker.sentinel.other
+        return_value = mocker.sentinel.return_value
+        patcher = mocker.patch.object(
             BaseMetadata, "combine", return_value=return_value
-        ) as mocker:
-            result = self.none.combine(other)
-
-        self.assertEqual(return_value, result)
-        self.assertEqual(1, mocker.call_count)
-        (arg,), kwargs = mocker.call_args
-        self.assertEqual(other, arg)
-        self.assertEqual(dict(lenient=None), kwargs)
-
-    def test_lenient(self):
-        other = sentinel.other
-        lenient = sentinel.lenient
-        return_value = sentinel.return_value
-        with mock.patch.object(
+        )
+        result = self.none.combine(other)
+
+        assert result == return_value
+        assert patcher.call_count == 1
+        (arg,), kwargs = patcher.call_args
+        assert arg == other
+        assert kwargs == dict(lenient=None)
+
+    def test_lenient(self, mocker):
+        other = mocker.sentinel.other
+        lenient = mocker.sentinel.lenient
+        return_value = mocker.sentinel.return_value
+        patcher = mocker.patch.object(
             BaseMetadata, "combine", return_value=return_value
-        ) as mocker:
-            result = self.none.combine(other, lenient=lenient)
+        )
+        result = self.none.combine(other, lenient=lenient)
 
-        self.assertEqual(return_value, result)
-        self.assertEqual(1, mocker.call_count)
-        (arg,), kwargs = mocker.call_args
-        self.assertEqual(other, arg)
-        self.assertEqual(dict(lenient=lenient), kwargs)
+        assert result == return_value
+        assert patcher.call_count == 1
+        (arg,), kwargs = patcher.call_args
+        assert arg == other
+        assert kwargs == dict(lenient=lenient)
 
-    def test_op_lenient_same(self):
+    def test_op_lenient_same(self, mocker):
         lmetadata = self.cls(**self.values)
         rmetadata = self.cls(**self.values)
         expected = self.values
 
-        with mock.patch("iris.common.metadata._LENIENT", return_value=True):
-            self.assertEqual(expected, lmetadata.combine(rmetadata)._asdict())
-            self.assertEqual(expected, rmetadata.combine(lmetadata)._asdict())
+        mocker.patch("iris.common.metadata._LENIENT", return_value=True)
+        assert expected == lmetadata.combine(rmetadata)._asdict()
+        assert expected == rmetadata.combine(lmetadata)._asdict()
```

Comment:
```suggestion
        assert lmetadata.combine(rmetadata)._asdict() == expected
        assert rmetadata.combine(lmetadata)._asdict() == expected
```

**Note:** there are a whole bunch more of these in this file.
I started by adding a suggestion for each, but it got totally out of hand, so I have deleted the others.
Easily found, wherever we have "expected =="

Reply:
ugh ...